KueuePopulator lock for helm charts complicates the release steps

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

When following the release script I hit this issue:

```
❯ make artifacts IMAGE_REGISTRY=registry.k8s.io/kueue GIT_TAG=$VERSION
if [ -d artifacts ]; then rm -rf artifacts; fi
DEST_CHART_DIR="artifacts" \
HELM="/usr/local/google/home/michalwozniak/go/src/sigs.k8s.io/kueue/bin/helm" YQ="/usr/local/google/home/michalwozniak/go/src/sigs.k8s.io/kueue/bin/yq" GIT_TAG="v0.15.2" IMAGE_REGISTRY="registry.k8s.io/kueue" \
HELM_CHART_PUSH= \
./hack/helm-chart-package.sh
Successfully packaged chart and saved it to: artifacts/kueue-0.15.2.tgz
Error: the lock file (Chart.lock) is out of sync with the dependencies file (Chart.yaml). Please update the dependencies
make: *** [Makefile:317: helm-chart-package] Error 1

```
and I need to manually remove the Chart.lock file


**What you expected to happen**:

No need to manually remove it

**How to reproduce it (as minimally and precisely as possible)**:

Follow the release steps.

--- a/hack/helm-chart-package.sh
+++ b/hack/helm-chart-package.sh
@@ -76,6 +76,8 @@ ${YQ}  e  ".kueueViz.frontend.image.repository = \"${IMAGE_REGISTRY}/kueueviz-fr
 ${YQ}  e  ".kueuePopulator.image.repository = \"${IMAGE_REGISTRY}/kueue-populator\" | .kueuePopulator.image.tag = \"${GIT_TAG}\" | .kueuePopulator.image.pullPolicy = \"IfNotPresent\"" -i cmd/experimental/kueue-populator/charts/kueue-populator/values.yaml
 
 # TODO: consider signing it
+rm -R cmd/experimental/kueue-populator/charts/kueue-populator/charts
+rm cmd/experimental/kueue-populator/charts/kueue-populator/Chart.lock
 ${HELM} package --version "${chart_version}" --app-version "${GIT_TAG}" charts/kueue -d "${DEST_CHART_DIR}"
 ${HELM} dependency build cmd/experimental/kueue-populator/charts/kueue-populator
 ${HELM} package --version "${chart_version}" --app-version "${GIT_TAG}" cmd/experimental/kueue-populator/charts/kueue-populator -d "${DEST_CHART_DIR}"

